### DISCUSSION TITLE
Mosaic statistics in a custom projection

### DISCUSSION THREAD

**Discussion by bmcandr** [stac-utils/titiler-pgstac#100] [Q&A] [answered]
Hi,

I would like to calculate statistics for a mosaic in a custom projection (e.g., an equal area projection like EPSG:6933). I suspect this is not possible without modifications similar to those suggested in #591?

**Comment by vincentsarago** [comment, marked as answer]
Hi @bmcandr 

In theory the mosaic backend `feature()` method accept both dst-crs and shape-crs options

https://github.com/stac-utils/titiler-pgstac/blob/d29a3247649753998cbf6fbb83e736169763afdc/titiler/pgstac/mosaic.py#L344-L345

but they are not surfaced at the API level 

https://github.com/stac-utils/titiler-pgstac/blob/d29a3247649753998cbf6fbb83e736169763afdc/titiler/pgstac/factory.py#L874-L914

2 solutions:
- add a `dst-crs` option 
- add a `shape-crs` option but then you'll need to provide a geojson in a custom projection  

both are available in titiler.core https://github.com/developmentseed/titiler/blob/main/src/titiler/core/titiler/core/dependencies.py#L454-L479


you can find examples of how to extend MosaicTilerFactories in https://stac-utils.github.io/titiler-pgstac/advanced/geojson_crop/ (note: the doc is not up to date and might try to import old functions)

**Reply by bmcandr** [reply]
Ah, I missed that `DstCRSParams` was already defined in the core. I had already implemented solution \# 1 by adding near-identical dependency to the `geojson_crop` route and it works reasonably well.

That said, in order to ensure the cropped mosaic maintains the native resolution of the data I found that I had to expose `width` and `height` parameters as well. I then compute those based on the resolution of the underlying data and include them in the request to the tiler. Does that sound right?

PS Great job on this and all of the other tools you've shared with the community! Thanks for everything.

**Reply by vincentsarago** [reply]
> Ah, I missed that DstCRSParams was already defined in the core. I had already implemented solution # 1 by adding near-identical dependency to the geojson_crop route and it works reasonably well.

it was added last week 🙈 

> That said, in order to ensure the cropped mosaic maintains the native resolution of the data I found that I had to expose width and height parameters as well. I then compute those based on the resolution of the underlying data and include them in the request to the tiler. Does that sound right?

Yeah it's always better to provide the height/width if you know it. I've always wondered about adding a `resolution` option but it will make things a bit more complex. 

maybe it's because of the default `max_size` https://github.com/stac-utils/titiler-pgstac/blob/main/titiler/pgstac/mosaic.py#L346 which will be ignored when providing height/width

**Reply by bmcandr** [reply]
Well that explains why I didn't see `DstCRSParams`!

This is very helpful information. I'm glad to know I'm on the right track.
